src/com/dotcms/plugin/saml/v3/init/DefaultInitializer.java

```diff
@@ -35,10 +53,41 @@
 
     private final AtomicBoolean initDone = new AtomicBoolean(false);
     private final SiteConfigurationParser siteConfigurationParser = new SiteConfigurationParser();
+    private final String hostVariableName = "Host";
+
+    private final StructureAPI structureAPI;
+    private final UserAPI userAPI;
+
+    public DefaultInitializer(){
+
+        this(APILocator.getStructureAPI(), APILocator.getUserAPI());
+    }
+
+    @VisibleForTesting
+    public DefaultInitializer(StructureAPI structureAPI, UserAPI userAPI) {
+
+        this.structureAPI = structureAPI;
+        this.userAPI = userAPI;
+    }
 
     @Override
     public void init(final Map<String, Object> context) {
 
+        Logger.info(this, "About to create SAML field under Host Content Type");
+        this.createSAMLField();
+        SamlHostPostHook postHook = new SamlHostPostHook();
+        Interceptor interceptor = (Interceptor)APILocator.getContentletAPIntercepter();
+        interceptor.delPostHookByClassName(postHook.getClass().getName());
+        try {
+            interceptor.addPostHook(postHook);
+        } catch (InstantiationException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
```

Comment:
we need to change this. e.printStackTrace() for better logging, and have only one catch | muticatch if the catch block will be the same